### DISCUSSION TITLE
Unable to print JSX to screen using Next.js

### DISCUSSION THREAD

**Discussion by richleach** [vercel/next.js#26472] [Help]
### What version of Next.js are you using?

11

### What version of Node.js are you using?

14.16.0

### What browser are you using?

Chrome

### What operating system are you using?

Mac 10.14.6

### How are you deploying your application?

 'npm run dev'

### Describe the Bug

I bet this is something I don't yet know how to do with Next.js but in the past with my React site I would retrieve data from an API that was in a regular Text field, with no markup (no Markdown) just plain text. I would write my content for that field locally, in text but write it in HTML, and when I needed to display some JSX code I would simply wrap that code in curly braces and tick marks. Then I'd finish marking the rest of the field up in HTML, save it in the DB and call it and it would display inside my React component when needed. However I just can't find a way to retrieve text called from an API that has embedded JSX that will display with the formatting respected. Is this not permitted in Next.js, or is there an unpublished convention on how to do this? Thank you in advance, I appreciate your help. Rich

### Expected Behavior

JSX code will be displayed, even slightly formatted from with my component's JSX.

### To Reproduce

Nothing I've tried has worked so far, I've tried using plain text fields, formatted text fields (Markdown, not using Markdown) and various escaping methods inside my JSX.

**Comment by yunghzz35** [comment]
I am not quite clear about what you are asking, but if you want to display a code snippet you can wrap your "plain text" with
```html
<code><pre><!--Your "plain text"--></pre></code>
```
p.s: next time please include your code in the "To Reproduce" section. That could help me understand more about the issue
### nextjs jsx code snippet example
```javascript
import Head from 'next/head'
import Image from 'next/image'
import styles from '../styles/Home.module.css'

export default function Home() {
  return (
    <div className={styles.container}>
      <Head>
        <title>Create Next App</title>
        <meta name="description" content="Generated by create next app" />
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <main className={styles.main}>
        <h1 className={styles.title}>
          Welcome to <a href="https://nextjs.org">Next.js!</a>
        </h1>
        {/*The following jsx code snippet in "text but write it in HTML" is correctly displayed on html with formatting*/}
        <code>
          <pre>
            {`
            const HelloWorld = (props) => {
              return <div>Hello World</div>
            }
            export default HelloWorld;
          `}
          </pre>
        </code>
        <p className={styles.description}>
          Get started by editing{' '}
          <code className={styles.code}>pages/index.js</code>
        </p>

        <div className={styles.grid}>
          <a href="https://nextjs.org/docs" className={styles.card}>
            <h2>Documentation &rarr;</h2>
            <p>Find in-depth information about Next.js features and API.</p>
          </a>

          <a href="https://nextjs.org/learn" className={styles.card}>{/*all 4 corners are round*/}
            <h2>Learn &rarr;</h2>
            <p>Learn about Next.js in an interactive course with quizzes!</p>
          </a>

          <a
            href="https://github.com/vercel/next.js/tree/master/examples"
            className={styles.card}
          >
            <h2>Examples &rarr;</h2>
            <p>Discover and deploy boilerplate example Next.js projects.</p>
          </a>

          <a
            href="https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
            className={styles.card}
          >
            <h2>Deploy &rarr;</h2>
            <p>
              Instantly deploy your Next.js site to a public URL with Vercel.
            </p>
          </a>
        </div>
      </main>

      <footer className={styles.footer}>
        <a
          href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
          target="_blank"
          rel="noopener noreferrer"
        >
          Powered by{' '}
          <span className={styles.logo}>
            <Image src="/vercel.svg" alt="Vercel Logo" width={72} height={16} />
          </span>
        </a>
      </footer>
    </div>
  )
}

```

**Reply by richleach** [reply]
![image](https://user-images.githubusercontent.com/1168807/122931535-3834be80-d32a-11eb-9f36-ef6baf0822c3.png)

I don't know how to get my React JSX to display just the un-rendered code. 
I apologize that this is so confusing, these editors are actually making my case for me, but they're clouding the issue at the same time.
Rich

**Reply by richleach** [reply]
BTW My approach for these pages is to add all of the component's return( *content* ) method to the database (CMS) where * content * is my actual page's content, which will include markup and additional JSX code (to be displayed as content, not necessarily to be used to drive this component) included in my blog post. For example: 

![image](https://user-images.githubusercontent.com/1168807/122955883-c1a1bc00-d33d-11eb-8a7e-132089ab5dae.png)


Is there a more appropriate way React/Next.js bloggers are building their blog websites? 

Thank you again for your help.

Rich

**Reply by richleach** [reply]
BTW your code works just fine - when I type the code directly into my component. I'm not doing that here, I'm pulling the values from a DB/CMS which are being stored in either Text or Rich text, causing my issue. What I'm asking is how do I store the JSX code in the DB/CMS then retrieve and display in my component?
